### ISSUE TITLE
Probabilistic Regression Tutorial: why do we need to manually shrink the value of the scale parameter

### ISSUE THREAD

**Issue by melantha-wang** [tensorflow/probability#1637]
Hi All,

I've been studying the [Probabilitistic Regression tutorial](https://www.tensorflow.org/probability/examples/Probabilistic_Layers_Regression) on the tensorflow-probability documentation. One thing I noticed is that often we need to manually shrink the value of the scale parameter. For example, in Case 4 of this tutorial,
```
# Build model.
model = tf.keras.Sequential([
  tfp.layers.DenseVariational(1 + 1, posterior_mean_field, prior_trainable, kl_weight=1/x.shape[0]),
  tfp.layers.DistributionLambda(
      lambda t: tfd.Normal(loc=t[..., :1],
                           scale=1e-3 + tf.math.softplus(0.01 * t[...,1:]))),
])
```
we have this `0.01 *` in the code, which, if removed, would cause the model to go crazy (image below is produced by running the same code without the `0.01 *`:
<img width="712" alt="image" src="https://user-images.githubusercontent.com/51358773/196682670-8ef4c3e0-4f9d-4f24-99f8-3b0f3df29451.png">

Intuitively it makes sense to me that we do not want a large scale parameter, which would make the model undesirably variable. But theoretically, if we give the model enough training time it should be able to figure out the right range of numbers, so why is this necessary? I would appreciate any insights or references/documentation on why we need to do this.

Thanks!

**Comment by Frightera** [comment]
> But theoretically, if we give the model enough training time it should be able to figure out the right range of numbers

Absolutely, it should converge to a better solution if you train it longer. I expected Adam optimizer to converge better than the screenshot. Maybe some other sophisticated optimizers (Rectified Adam, Adabelief) can provide better results.

> So why is this necessary?

I *guess*, it speeds up the optimization, firstly you want to focus on one factor here because the model needs to learn both mean and scale. When it is multiplied by 0.01, the effects of gradients will be reduced. Therefore model will not face with extreme values during optimization.

Actually, the key thing here is that, multiplying it by 0.01 will not change the space of possible values of the scale. This can easily be proved by using an adaptive optimizer and training the model for 8000-9000 epochs. The output should be similar to the given example where it is multiplied by 0.01

**Comment by csuter** [comment]
I think it's sometimes (often? always?) better to start variational
surrogates with small scale and let it grow as needed. Probably there's a
local optimum where's the scale is large, all variation is attributed to
noise, and no other structure is able to be discerned?

**Comment by melantha-wang** [comment]
Thank you both for your comments which make a lot of sense!
